content/blog/using-the-hpegl-hpe-greenlake-and-morpheus-terraform-providers-in-concert-to-create-provision-and-manage-hpe-pce-private-cloud-enterprise-vmaas-instances.md

```diff
@@ -0,0 +1,1492 @@
+
+* A Morpheus Access Token 
+* The URL of the Morpheus instance 
+* The time to expiry of the Access Token (in seconds) 
+
+The first two outputs can then be passed into a Morpheus provider stanza.  See the following example where there is just one *location* for a VMaaS instance:
+
+```hcl
+provider "hpegl" {
+  vmaas {
+    location   = var.location
+    space_name = var.space
+  }
+}
+
+data "hpegl_vmaas_morpheus_details" "morpheus_details" {}
+
+provider "morpheus" {
+  url          = data.hpegl_vmaas_morpheus_details.morpheus_details.url
+  access_token = data.hpegl_vmaas_morpheus_details.morpheus_details.access_token
+}
+```
+
+For VMaaS instances with multiple locations *Provider Aliasing* can be used to tie a specific VMaaS *location* to the corresponding Morpheus instance:
+
+```hcl
+# Location 1
+provider "hpegl" {
+  vmaas {
+    location   = var.location_1
+    space_name = var.space_1
+  }
+
+  alias = "location_1"
+}
+
+data "hpegl_vmaas_morpheus_details" "location_1" {
+  provider = hpegl.location_1
+}
+
+provider "morpheus" {
+  url          = data.hpegl_vmaas_morpheus_details.location_1.url
+  access_token = data.hpegl_vmaas_morpheus_details.location_1.access_token
+
+  alias = "morpheus_location_1"
+}
+
+
+# Location 2
+provider "hpegl" {
+  vmaas {
+    location   = var.location_2
+    space_name = var.space_2
+  }
+
+  alias = "location_2"
+}
+
+data "hpegl_vmaas_morpheus_details" "location_2" {
+  provider = hpegl.location_2
+}
+
+provider "morpheus" {
+  url          = data.hpegl_vmaas_morpheus_details.location_2.url
+  access_token = data.hpegl_vmaas_morpheus_details.location_2.access_token
+
+  alias = "morpheus_location_2"
+}
+```
+
+# **Complete examples for One VMaaS Location**
+
+In the next two sections we present complete HCL (Hashicorp Configuration Language) examples for two configurations each with one VMaaS Location: 
+
+* One VMaaS Cloud, with one VMaaS instance 
+* Two VMaaS Clouds, each with one VMaaS instance for a total of two VMaaS instances 
+
+ The clouds are vSphere clouds.  In both cases: 
+
+* hpegl is used to: 
+
+  * Define the VMaaS Location and Space 
+  * Retrieve the relevant Morpheus Token and URL 
+  * Retrieve details for the VMaaS Datastore, Network, Resource Pool, Plan, Environment and Cloud Folder 
+  * Create a VMaaS instance 
+* Morpheus is used to: 
+
+  * Retrieve details for the VMaaS Cloud 
+  * Create a Group for the above cloud 
+  * Create an Instance Type 
+  * Retrieve details for a VM Image 
+  * Create a Node Type for the VM Image 
+  * Create an Instance Layout for the above Node Type and Instance Type 
+
+## **One Location/One VMaaS Cloud/One VMaaS Instance example**
+
+![](/img/hpegl-and-morpheus-terraform-blog-post-frame-1-1-.jpg)
+
+### HCL
+
+Here is an example piece of HCL (Hashicorp Configuration Language) that uses hpegl and the Morpheus provider in the case of One VMaaS Cloud:
```

Comment:
Suggestion:

Here is an example piece of HCL that uses hpegl and the Morpheus provider in the case of One VMaaS Cloud: